lib/server.js

```diff
@@ -24,13 +24,27 @@ function handleResponse (opts, req, resp, response) {
             response.headers = {};
         }
 
+        var rh = response.headers;
+
         // Set up CORS
-        response.headers['Access-Control-Allow-Origin'] = '*';
-        response.headers['Access-Control-Allow-Methods'] = 'GET';
-        response.headers['Access-Control-Allow-Headers'] = 'accept, content-type';
+        rh['Access-Control-Allow-Origin'] = '*';
+        rh['Access-Control-Allow-Methods'] = 'GET';
+        rh['Access-Control-Allow-Headers'] = 'accept, content-type';
+
+        if (!/^application\/json/.test(rh['content-type'])) {
+            // Set up security headers
+            // https://www.owasp.org/index.php/List_of_useful_HTTP_headers
+            rh['X-XSS-Protection'] = '1; mode=block';
+            rh['X-Content-Type-Options'] = 'nosniff';
+            rh['X-Frame-Options'] = 'SAMEORIGIN';
+            rh['Content-Security-Policy'] =
+                "default-src 'none'; media-src *; img-src *; style-src *";
```

Comment:
> So for non json content, is there any case not covered by 'self' and *.wikimedia.org for style/img/media?

For our purposes that would probably be okay, but it definitely needs to be configurable. For example, it needs to be different in labs, staging & for all third-party users. That's why I said that we can do this in the next step, rather than right away.